web/src/settings_ui.ts

```diff
@@ -55,6 +55,16 @@ export function do_settings_change(
         url,
         data,
         success(response_data) {
+            if (
+                typeof response_data === "object" &&
+                response_data &&
+                "msg" in response_data &&
+                typeof response_data.msg === "string" &&
+                response_data.msg
+            ) {
+                success_msg_html = response_data.msg;
+            }
```

Comment:
Instead of modifying this function which is used by many settings, would it be better to just handle default language setting as a special case to show the success response from server since for all other settings we do not have any success response from server.

Reply:
Thanks for the feedback and sorry for the delay.

I think the `do_settings_change` function has to be changed in some way because that is the only place where `response_data` is available as far as I know, but I agree it would be better to treat default language as a special case.

In the new commit (4488e78f1ccd8d34454a248a817d53b90195a50f), I added a new option `success_msg_html_generator` to the `do_settings_change` function to generate a success message from the response. Currently, `success_msg_html_generator` is only used when `default_language` is part of the settings to be updated. Is this okay?

Reply:
I was thinking of directly using `channel.patch` instead of using `do_settings_change` or `change_display_setting`.

Reply:
Thanks for the clarification! I reverted changes made to `do_settings_change` and `change_display_setting` and added a new function to handle default language setting as a special case (see 99c07f075567bda665209615bdfac7fb26f762b6 and ade85f9659012eb4cce8757de54d6f3e0d6d339a).